### ISSUE TITLE
Migration from self-hosted Postgres to LHDI RDS

### ISSUE THREAD

**Issue by msnwatson** [department-of-veterans-affairs/abd-vro#1845]
**User Story**

As a VRO engineer, I want to leverage the upcoming offering of RDS from LHDI to reduce maintenance and overhead of hosting our own Postgres instance.

**Acceptance Criteria**

1. All database tables/schemas across all LHDI environments are migrated to RDS 
2. Wiki references to our self-hosted Postgres instances are updated/deprecated as necessary to reflect the move to RDS
3. Self-hosted postgres container can be removed from docker-compose specs, helm charts, and GitHub actions (caveat: maintain some instances for local development environment).
4. Migration should be tested

**Dependencies**
This ticket cannot be started until LHDI releases the RDS offering to customer teams. Based on, https://dsva.slack.com/archives/C03UA9MV1EH/p1690214400718919 this should be happening by at least the end of the first week of August.

**Notes about work**
- To achieve AC 1, we should be able to leverage our Flyway integration
- Slack thread with suggestions for how to get started https://dsva.slack.com/archives/C04QLHM9LR0/p1690223816692669
- This will invalidate the work needed for https://github.com/department-of-veterans-affairs/abd-vro/issues/1096

**Comment by yoomlam** [comment]
Related to #1119, which _may_ be easily accomplished as part of this ticket, i.e., by setting backup RDS configurations.

**Comment by yoomlam** [comment]
Should also consider how easily Partner Teams can create and maintain their own DB.

**Comment by dianagriffin** [comment]
Note from backlog refinement discussion re: maintaining an option for local development, the postgres stuff could be moved to mocks.

**Comment by dianagriffin** [comment]
There are two parts to this work:
- There's effort related to "don't use the self-hosted db" 
- Another effort related to "get the self-hosted db to work locally"

Both should wait until RDS is available so that we are informed by their docs

**Comment by pshahVA** [comment]
https://lighthouseva.slack.com/archives/C03UA9MV1EH/p1690215413872109?thread_ts=1690214400.718919&cid=C03UA9MV1EH

**Comment by dianagriffin** [comment]
Note from backlog refinement discussion 8/10: Please confirm that the legacy database schemas from the RRD product have been deprecated and are not being replicated in this migration.

**Comment by yoomlam** [comment]
> Note from backlog refinement discussion 8/10: Please confirm that the legacy database schemas from the RRD product have been deprecated and are not being replicated in this migration.

Confirmed. There is no data from RRD in the current VRO DB.

**Comment by msnwatson** [comment]
RDS instances are created, but I am struggling to find a way to interact with those instances because the instances have strict authorization policies for different actions. I am continuing to work toward a solution there.

I have already done some work to remove postgres from our remote deployments on a local branch but will hold off on creating a PR until I feel comfortable in our RDS interactivity in the remote environments.

**Comment by msnwatson** [comment]
I made updates to https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/AWS-RDS-Integration to reflect the procedures followed to produce a working RDS dbinstance in the QA nonprod environment. RDS instances exist in all of our LHDI environments but are only partially initialized outside of the QA namespace.

**Comment by msnwatson** [comment]
I have 2 PRs out right now and had originally intended on merging them quite quickly; however, I ran into some issues with deployments to try to create the Flyway migrations on our RDS instances. Now, I will need to test these PRs more rigorously, especially https://github.com/department-of-veterans-affairs/abd-vro/pull/2040

**Comment by yoomlam** [comment]
I didn't look into the details of your roadbumps, but there's little coupling of VRO to Flyway, so it can be replaced with whatever works better with our connection to RDS.

**Comment by nelsestu** [comment]
The pull request is now showing that it is ready for squash+merge: https://github.com/department-of-veterans-affairs/abd-vro/pull/2040#pullrequestreview-1649606088

**Comment by msnwatson** [comment]
This was originally blocked by #1007 , however now that this has been addressed there is another issue represented by #2180 which would make deploying the changes to completely mark this ticket as done break the `svc-bip-api` and `svc-lighthouse-api` containers.

We will attempt to address #2180 unless a pressing need emerges that requires RDS over the functionality of the two microservices.

**Comment by raudabaugh** [comment]
Hi VRO Team! I'm a new Benefits Portfolio engineer helping the EE and CC teams, and I'd like to better understand how the migration to RDS will impact us. Here are some questions we're thinking about as we start to design new SQL-dependent features:

1, Can we assume that any flyway migrations placed in `db-init/src/main/resources/database/migrations` will be available in RDS?
2. Will the self-hosted postgres container continue to be the preferred method for local development/testing of SQL changes?
3. Should domain microservices like EE and CC use the `vro` database or create a new database on the same RDS instance?
4. If domain microservices are to use the `vro` database, should they create a separate user for access, or use one of the existing vro ones?
5. If domain microservices are to use the `vro` database, should they keep their SQL changes in a separate schema?

Any guidance would be appreciated - thanks!

**Comment by dianagriffin** [comment]
Hi @raudabaugh! I don't think this answers all your questions, but wanted to surface our [VRO Database](https://github.com/department-of-veterans-affairs/abd-vro/wiki/VRO-Database) documentation in case you haven't seen it yet. I'll defer to @msnwatson to answer any of your questions that aren't yet addressed in those docs!
